svc.go

```diff
@@ -188,28 +247,52 @@ func (as *AutoNATService) doDial(pi peer.AddrInfo) *pb.Message_DialResponse {
 		return newDialResponseError(pb.Message_E_DIAL_ERROR, "dial failed")
 	}
 
-	conns := as.dialer.Network().ConnsToPeer(pi.ID)
-	if len(conns) == 0 {
-		log.Errorf("supposedly connected to %s, but no connection to peer", pi.ID.Pretty())
-		return newDialResponseError(pb.Message_E_INTERNAL_ERROR, "internal service error")
-	}
-
-	ra := conns[0].RemoteMultiaddr()
+	ra := conn.RemoteMultiaddr()
 	as.dialer.Network().ClosePeer(pi.ID)
 	return newDialResponseOK(ra)
 }
 
+func (as *AutoNATService) enableWhenPublic() {
```

Comment:
Please can you add a unit test that emits mock Reachability events on the bus & ensures this function works as expected ? We should test both the  `SetStreamHandler ` &  `RemoveStreamHandler` branches.